pkgs/test_core/lib/src/runner/vm/platform.dart

```diff
@@ -48,23 +48,53 @@ class VMPlatform extends PlatformPlugin {
 
     _setupPauseAfterTests();
 
-    var receivePort = ReceivePort();
+    MultiChannel outerChannel;
+    var cleanupFns = <void Function()>[];
     Isolate? isolate;
-    try {
-      isolate = await _spawnIsolate(
-          path, receivePort.sendPort, suiteConfig.metadata, platform.compiler);
-      if (isolate == null) return null;
-    } catch (error) {
-      receivePort.close();
-      rethrow;
+    if (platform.compiler == Compiler.exe) {
+      var serverSocket = await ServerSocket.bind('localhost', 0);
+      Process process;
+      try {
+        process =
+            await _spawnExecutable(path, suiteConfig.metadata, serverSocket);
+      } catch (error) {
+        serverSocket.close();
+        rethrow;
+      }
+      process.stdout.map(stdout.add);
+      process.stderr.map(stderr.add);
+      var socket = await serverSocket.first;
+      outerChannel = MultiChannel<Object?>(StreamChannel(socket, socket)
```

Comment:
Maybe add a utility for this since it is symmetric?

Reply:
Done